packages/lesswrong/server/collections/ultraFeedEvents/mutations.ts

```diff
@@ -35,9 +46,45 @@ export async function createUltraFeedEvent({ data }: CreateUltraFeedEventInput,
   return document;
 }
 
+export async function updateUltraFeedEvent(args: { selector: SelectorInput, data: CreateUltraFeedEventDataInput }, context: ResolverContext) {
+  const { selector, data: inputData } = args;
+  const { UltraFeedEvents } = context;
+
+  const documentSelector = convertDocumentIdToIdInSelector(selector as UpdateSelector);
+  const existingDoc = await context.loaders.UltraFeedEvents.load(documentSelector._id);
+  
+  if (!existingDoc) {
+    throw new Error('UltraFeedEvent not found');
+  }
+  
+  // Validate text length if updating a seeLess event with text feedback
+  if ('event' in inputData && inputData.event?.feedbackReasons?.text) {
+    const textLength = inputData.event.feedbackReasons.text.length;
+    if (textLength > 500) {
+      throw new Error('Feedback text must be 500 characters or less');
+    }
+  }
+  
+  const updateData: Partial<DbUltraFeedEvent> = {};
+  if ('event' in inputData && inputData.event !== undefined) {
+    updateData.event = inputData.event;
+  }
```

Comment:
I don't quite understand what this is doing; I think if you use the `UpdateUltraFeedEventDataInput` type this won't be necessary since it'll be restricted to only the field(s) that you're allowed to update (which is just the `event` field).

Reply:
(Actually, speaking of which, you need to update the schema to make the `event` field updatable... I don't know how any of this ever worked without that, but it shouldn't have, and if it did that reveals a pretty serious bug.)